packages/design-toolkit/src/components/flashcard/flashcard.test.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { uuid } from '@accelint/core';
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
+import { Fragment } from 'react/jsx-runtime';
+import { describe, expect, it } from 'vitest';
+import {
+  Flashcard,
+  FlashcardAdditionalData,
+  FlashcardDetailLabel,
+  FlashcardDetailsList,
+  FlashcardDetailValue,
+  FlashcardHeader,
+  FlashcardHero,
+  FlashcardSubheader,
+} from '.';
+
+function setup(isLoading = false) {
+  const details = [
+    { label: 'OBJECTID', value: 75, id: uuid() },
+    { label: 'NAME', value: 'Acrisure Stadium', id: uuid() },
+    { label: 'LATITUDE', value: '40-26-48.22N', id: uuid() },
+    { label: 'LONGITUDE', value: '080-00-56.66W', id: uuid() },
+    { label: 'CITY', value: 'Pittsburgh', id: uuid() },
+    { label: 'STATE', value: 'PA', id: uuid() },
+    { label: 'STATUS_CODE', value: 'Open', id: uuid() },
+    { label: 'OPENING_ON', value: null, id: uuid() },
+  ];
+
+  render(
+    <Flashcard isLoading={isLoading}>
+      <FlashcardHero data-testid='hero'>
+        <FlashcardHeader>IDENTIFIER</FlashcardHeader>
+        <FlashcardSubheader>DATA</FlashcardSubheader>
+      </FlashcardHero>
+      <FlashcardAdditionalData>SECONDARY_DATA_01</FlashcardAdditionalData>
+      <FlashcardAdditionalData>SECONDARY_DATA_02</FlashcardAdditionalData>
+      <FlashcardDetailsList data-testid='secondary'>
+        {details.map((detail) => (
+          <Fragment key={detail.id}>
+            <FlashcardDetailLabel>{detail.label}</FlashcardDetailLabel>
+            <FlashcardDetailValue>{detail.value}</FlashcardDetailValue>
+          </Fragment>
+        ))}
+      </FlashcardDetailsList>
+    </Flashcard>,
+  );
+}
+
+describe('Flashcard', () => {
+  it('should render', () => {
+    setup();
+    expect(screen.getByText('IDENTIFIER')).toBeDefined();
+    expect(screen.getByText('DATA')).toBeDefined();
+    expect(screen.getByText('SECONDARY_DATA_01')).toBeDefined();
+    expect(screen.getByText('SECONDARY_DATA_02')).toBeDefined();
+  });
+
+  it('should only show 5 additional details', () => {
+    setup();
+
+    // First item to be hidden by selector nth-of-type
+    const detailEntries = document.getElementsByClassName(
+      'nth-of-type-[n+6]:hidden',
+    );
```

Comment:
I think there's a much easier way to test this - you could instead look at what is actually rendered on the screen and test that the labels of the items past 5 .not.toBeInTheDocument()

generally speaking, you shouldn't be using document queries in your tests. Instead you should stick to the RTL queries off of screen.

For this one, since you're asserting something is not in the document, it should be `expect(screen.queryByText('STATE')).not.toBeInTheDocument()`